src/rgw/rgw_metadata.cc

```diff
@@ -344,20 +340,312 @@ class RGWMetadataLogOmap final : public RGWMetadataLogBE {
+      return r;
+    }
+    for (const auto& entry : log_entries) {
+      bufferlist& bl((bufferlist &)entry);
+      cls_log_entry log_entry;
+      decode(log_entry, bl);
+/*      log_entry.id = entry.marker;
+      log_entry.timestamp = entry.mtime;
+      auto liter = entry.data.cbegin();
+      try {
+	decode(log_entry.entry, liter);
+      } catch (const buffer::error& err) {
+	lderr(cct) << __PRETTY_FUNCTION__
+		   << ": failed to decode data changes log entry: "
+		   << err.what() << dendl;
+	return -EIO;
+      }*/
+      e.push_back(std::move(log_entry));
+    }
+    if (truncated)
+      *truncated = more;
+    if (out_marker && !log_entries.empty()) {
+      *out_marker = log_entries.back().marker;
+    }
+    return 0;
+  }
+  int get_info(int index, RGWMetadataLogInfo *info) override {
+    auto& fifo = fifos[index];
+    auto r = fifo->read_meta(null_yield);
+    if (r < 0) {
+      lderr(cct) << __PRETTY_FUNCTION__
+		 << ": unable to get FIFO metadata: " << get_shard_oid(index)
+		 << ": " << cpp_strerror(-r) << dendl;
+      return r;
+    }
+    auto m = fifo->meta();
+    auto p = m.head_part_num;
+    if (p < 0) {
+      info->marker = rgw::cls::fifo::marker{}.to_string();
+      info->last_update = ceph::real_clock::zero();
+      return 0;
+    }
+    rgw::cls::fifo::part_info h;
+    r = fifo->get_part_info(p, &h, null_yield);
+    if (r < 0) {
+      lderr(cct) << __PRETTY_FUNCTION__
+		 << ": unable to get part info: " << get_shard_oid(index) << "/" << p
+		 << ": " << cpp_strerror(-r) << dendl;
+      return r;
+    }
+    info->marker = rgw::cls::fifo::marker{p, h.last_ofs}.to_string();
+    info->last_update = h.max_time;
+    return 0;
+  }
+
+  int get_info_async(int index, RGWMetadataLogInfoCompletion *completion) override {
+    auto& fifo = fifos[index];
+    auto r = fifo->read_meta(null_yield);
+    if (r < 0) {
+      lderr(cct) << __PRETTY_FUNCTION__
+		 << ": unable to get FIFO metadata: " << get_shard_oid(index)
+		 << ": " << cpp_strerror(-r) << dendl;
+      return r;
+    }
+    auto m = fifo->meta();
+    auto p = m.head_part_num;
+    cls_log_header& header(completion->get_header());
+    if (p < 0) {
+      header.max_marker = rgw::cls::fifo::marker{}.to_string();
+      header.max_time = utime_t(ceph::real_clock::zero());
+      return 0;
+    }
+    librados::AioCompletion* c = completion->get_completion();
+    rgw::cls::fifo::part_info h;
+    r = fifo->get_part_info(p, &h, c);
+    if (r < 0) {
+      lderr(cct) << __PRETTY_FUNCTION__
+		 << ": unable to get part info: " << get_shard_oid(index) << "/" << p
+		 << ": " << cpp_strerror(-r) << dendl;
+      return r;
+    }
+    c->wait_for_complete();
+    r = c->get_return_value();
+    if (r < 0) {
+      lderr(cct) << __PRETTY_FUNCTION__
+		 << ": AIO completion error for push of " << get_shard_oid(index)
+		 << ": " << cpp_strerror(-r) << dendl;
+    }
+    header.max_marker = rgw::cls::fifo::marker{p, h.last_ofs}.to_string();
```

Comment:
This shouldn't go here. It should go in RGWMetadataLogInfoCompletion.

(Or we could modify get_head_info to take aconverter function. RGWMetadataLogInfoCompletion seemed more natural since I thought we were already using that everywhere we'd call this function, but if not modifying get_head_info would be the obvious choice.)

Reply:
yes.. Looks like RGWMetadataLogInfoCompletion doesn't seem to be updating any values. Even in timelog get_info_async, completion->header is passed as an argument to be filled.

>>>>
    return cls.timelog.info_async(completion->get_io_obj(), oids[index],
                                       &completion->get_header(),
                                       completion->get_completion());
  }
<<<<

I am updating the branch with latest changes. Please take a look. Thanks!